### PR TITLE
Add search bars to courses list page, modules list page, lesson list page and quiz list page

### PR DESCRIPTION
**Pull request by manojvaghela**

### CODE REVIEW

File: yaksh/views.py [hunk 1]
```diff
@@ -1043,8 +1043,30 @@ def courses(request):
...
+            courses = courses.filter(
+                name__contains=course_tags)
+        elif course_status == 'active' :
+            courses = courses.filter(
+                name__contains=course_tags, active=True)
+        elif course_status == 'closed':
+            courses = courses.filter(
+                name__contains=course_tags, active=False)
+        context["courses_found"] = courses.count()
+    print("courses_found: ", context)
```
Comment (adityacp) [inline yaksh/views.py]: Remove print statements

File: yaksh/views.py [hunk 2]
```diff
@@ -2606,7 +2628,26 @@ def show_all_quizzes(request):
...
+        quiz_tags = request.POST.get('quiz_tags')
+        quiz_status = request.POST.get('quiz_status')
+
+        if quiz_status == 'select':
+            quizzes = quizzes.filter(description__contains=quiz_tags)
+        elif quiz_status == 'active':
+            quizzes = quizzes.filter(description__contains=quiz_tags, active=True)
+        elif quiz_status == 'closed':
+            quizzes = quizzes.filter(description__contains=quiz_tags, active=False)
+        context["quizzes_found"] = len(quizzes)
```
Comment (adityacp) [inline yaksh/views.py]: Do not use `len(queryset)` instead always use `queryset.count()`

File: yaksh/templates/yaksh/courses.html [hunk 3]
```diff
@@ -53,16 +53,47 @@
...
+                            <div class="row col-md-12">
+                                <h6>Filter Courses: </h6>
+                                <div name="course_status" id="course_status" style="margin-left: 20px;">
+                                    {{ form.course_status }}
+                                </div>
+                            </div>
+                            <br>
+                            <button class="btn btn-primary" type="submit">Search</button>
+
+                            <button class="btn btn-primary" type="button" onClick='location.replace("{{URL_ROOT}}");'>
```
Comment (CruiseDevice) [inline yaksh/templates/yaksh/courses.html]: Use django URL pattern in place of `location.replace()`

File: yaksh/templates/yaksh/courses.html [hunk 4]
```diff
@@ -53,16 +53,47 @@
...
+                                Clear Search
+                            </button>
+                        </div>
+                      </div>
+                    </div>
+                </form>
+                <br>
+                <div id="created_courses">
                 {% if not courses %}
```
Comment (CruiseDevice) [inline yaksh/templates/yaksh/courses.html]: This Jinja logic can be improved.

File: yaksh/templates/yaksh/courses.html [hunk 5]
```diff
@@ -334,13 +365,46 @@ <h4>{{course.creator.get_full_name.title}}</h4>
...
+                            <br>
+                            <div class="row col-md-12">
+                                <h6>Filter Quizzes: </h6>
+                                <div name="quiz_status" id="quiz_status" style="margin-left: 20px;">
+                                    {{ form.quiz_status }}
+                                </div>
+                            </div>
+                            <br>
+                            <button class="btn btn-primary" type="submit">Search</button>
+                            <button class="btn btn-primary" type="button" onClick='location.replace("{{URL_ROOT}}");'>
```
Comment (CruiseDevice) [inline yaksh/templates/yaksh/courses.html]: Replace `location.replace()` with Django URL pattern.

File: yaksh/templates/yaksh/courses.html [hunk 6]
```diff
@@ -334,13 +365,46 @@ <h4>{{course.creator.get_full_name.title}}</h4>
...
+                                Clear Search
+                            </button>
+                        </div>
+                      </div>
+                    </div>
+                </form>
+                <br>
+
                 <a href="{{URL_ROOT}}/exam/manage/addquiz/" class="btn btn-primary">Add New Quiz</a>
```
Comment (CruiseDevice) [inline yaksh/templates/yaksh/courses.html]: Replace href with Django URL pattern.

**Comment by adityacp** [comment]
Closing this PR because of updated PR #665

**Pull request closed without merging**